httpx/_utils.py

```diff
@@ -309,6 +310,37 @@ def unquote(value: str) -> str:
     return value[1:-1] if value[0] == value[-1] == '"' else value
 
 
+def guess_content_type(filename: typing.Optional[str]) -> typing.Optional[str]:
+    if filename:
+        return mimetypes.guess_type(filename)[0] or "application/octet-stream"
+    return None
+
+
+def get_filelike_length(stream: typing.IO) -> int:
```

Comment:
Looks good! Based off of an existing implementation someplace, or figured out from scratch?

Reply:
Basically a patchwork of info from here and there :-) eg the [io](https://docs.python.org/3/library/io.html) module docs, and SO for `os.fstat` and "how to get length of BytesIO/StringIO". :-)